packages/block-library/src/navigation/editor.scss

```diff
@@ -522,10 +522,40 @@ body.editor-styles-wrapper
 	}
 }
 
-.wp-block-navigation__unsaved-changes-warning {
-	width: 100%;
+@keyframes fadein {
+	0% {
+		opacity: 0;
+	}
+	100% {
+		opacity: 1;
+	}
+}
 
-	.block-editor-warning__actions {
-		margin-top: 0;
+.wp-block-navigation__unsaved-changes {
+	position: relative;
+
+	.components-spinner {
+		position: absolute;
+		top: calc(50% - #{$spinner-size} / 2);
+		left: calc(50% - #{$spinner-size} / 2);
+
+		// Delay showing the saving spinner until after 2 seconds.
+		// This should ensure it only shows for slow connections.
+		opacity: 0;
+		animation: 0.5s linear 2s 1 normal forwards fadein;
+	}
+}
+
+@keyframes fadeinhalf {
+	0% {
+		opacity: 1;
 	}
+	100% {
+		opacity: 0.5;
+	}
+}
+
+.wp-block-navigation__unsaved-changes-overlay.is-saving {
+	opacity: 1;
+	animation: 0.5s linear 2s 1 normal forwards fadeinhalf;
```

Comment:
Not sure if there's a better way to declare these animations.

Reply:
It looks fine! You shouldn't need to specify the iteration count though, because `1` is its default value.

Reply:
Cool, have updated that. Also noticed that this one fades out, so it should be called `fadeouthalf`. Corrected that too. 😄